quic/s2n-quic-core/src/recovery/bbr.rs

```diff
@@ -767,18 +808,24 @@ impl BbrCongestionController {
     //# while limiting the window to the larger of 14,720 bytes or twice the
     //# maximum datagram size.
     #[inline]
-    fn initial_window(max_datagram_size: u16) -> u32 {
+    fn initial_window(max_datagram_size: u16, app_settings: &ApplicationSettings) -> u32 {
         const INITIAL_WINDOW_LIMIT: u32 = 14720;
-        min(
+        let default = min(
             10 * max_datagram_size as u32,
             max(INITIAL_WINDOW_LIMIT, 2 * max_datagram_size as u32),
-        )
+        );
+        app_settings
+            .initial_congestion_window
+            .map(|initial_cwnd|
+                // prevent the application from setting a value that is too small
+                max(initial_cwnd, BbrCongestionController::minimum_window(max_datagram_size)))
```

Comment:
I'd say just enforce the minimum_window at the end, even if the initial window isn't overridden. It's possible that if the `max_datagram_size` is larger than the typical 1500, the default initial window will be too small

Reply:
Ahh good point!